clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/op/manifest/KubernetesDeployManifestOperation.java

```diff
@@ -167,18 +168,21 @@ public OperationResult operate(List<OperationResult> _unused) {
   @NotNull
   private List<KubernetesManifest> getManifestsFromDescription() {
     List<KubernetesManifest> inputManifests = description.getManifests();
-    if (inputManifests == null || inputManifests.isEmpty()) {
-      // The stage currently only supports using the `manifests` field but we need to continue to
-      // check `manifest` for backwards compatibility until all existing stages have been updated.
-      @SuppressWarnings("deprecation")
-      KubernetesManifest manifest = description.getManifest();
+
+    // The stage currently only supports using the `manifests` field but we need to continue to
+    // check `manifest` for backwards compatibility until all existing stages have been updated.
+    @SuppressWarnings("deprecation")
+    KubernetesManifest manifest = description.getManifest();
+
+    if (CollectionUtils.isEmpty(inputManifests) && manifest != null) {
```

Comment:
If both `description.getManifests()` and `description.getManifest()` return `null`, there will be a new `NullPointerException` in line 186, so we need to make sure that `inputManifests` is never `null`.